src/core/manager.c

```diff
@@ -3306,6 +3315,80 @@ static int manager_deserialize_units(Manager *m, FILE *f, FDSet *fds) {
         return 0;
 }
 
+usec_t manager_get_runtime_watchdog(Manager *m) {
+        assert(m);
+
+        if (MANAGER_IS_USER(m))
+                return USEC_INFINITY;
+
+        if (timestamp_is_set(m->runtime_watchdog_overridden))
+                return m->runtime_watchdog_overridden;
+
+        return m->runtime_watchdog;
+}
+
+usec_t manager_get_reboot_watchdog(Manager *m) {
+        assert(m);
+
+        if (timestamp_is_set(m->reboot_watchdog_overridden))
+                return m->reboot_watchdog_overridden;
+
+        return m->reboot_watchdog;
```

Comment:
Hmm, normally, we'd have a boolean flag like `reboot_watchdog_overridden` and just a single time value. Having those two time values seems overly complicated... After all, once overridden, we shouldn't refer to the old value ever again.

Reply:
I preferred not using a boolean flag otherwise we would have needed to save/restore the value while reloading PID1 configuration like we do currently for `log_level` and `log_target`, see https://github.com/systemd/systemd/blob/master/src/core/main.c#L1781, which is pretty ugly IMHO, especially if the number of PID1 settings grows.

OTOH using a second time value allows to transparently reload PID1 settings without the need of saving/restoring the overridden value as described previously.

Reply:
I would very much prefer to use the same scheme for all settings. So if the log settings use a flag, that is an even stronger argument to use it here too (or to convert all of them to the new pattern, but as I said, I think it is inferior).

Reply:
The log settings are using a flag because they're a bit "special" - they are currently not part of  the manager object itself but instead are managed via global variables. So they differ from the other settings.

Also if you want to use the same pattern, it might be a better idea to keep the one that can restore the old value because soon or later one of the manager property will need it.

In fact `ShowStatus` could be a good candidate.